### PR TITLE
Added max_intervals prop to Interval component

### PR DESCRIPTION
**Pull request by valentijnnieman**
As per #222.

edit: 
A quick example:
```
...
app.layout = html.Div([
    html.Div(id='output'),
    dcc.Interval(id='interval', interval=2000, max_intervals=2)
])


@app.callback(Output('output', 'children'),
              [Input('interval', 'n_intervals')])
def update_text(n):
    return "This page has been updated {} times, max_interval=2".format(n)
...
```

### CODE REVIEW

**Comment by rmarren1** [comment]
Is this supposed to automatically stop the interval from triggering the `n_intervals` property is `>=` `max_intervals`? This is the behavior I expected, but it does not seem to be working that way.
Using this code:
```
import dash
from dash.dependencies import Input, Output, State
import dash_core_components as dcc
import dash_html_components as html

app = dash.Dash('hello')


app.layout = html.Div([
    html.Div(id='output'),
    dcc.Interval(id='interval', interval=2000, max_intervals=2)
])



@app.callback(Output('output', 'children'),
              [Input('interval', 'n_intervals'),
               Input('interval', 'max_intervals')])
def update_text(n, M):
    return "This page has been updated {} times, max_intervals={}".format(n, M)


app.run_server(debug=True)

```
I get this:

image

**Comment by valentijnnieman** [comment]
@rmarren1 Thanks for reviewing! You need to set `scripts.config.serve_locally = True` in order for the local changes to work - otherwise Dash will get the components from unpkg, which won't have the changes I made! I will specify that in the example next time, left that stuff out to make it shorter.

**Comment by valentijnnieman** [comment]
Happens to me all the time too. That's a good point - I'll rewrite it to use `-1`! Thanks again.

**Comment by rmarren1** [comment]
Great, I think the docstring for `max_intervals` should explain that -1 means no maximum, but other than that I am :dancer:

File: src/components/Interval.react.js [hunk 1]
```diff
@@ -75,6 +81,11 @@ Interval.propTypes = {
      */
     n_intervals: PropTypes.number,
 
+    /**
+     * Number of times the interval will be fired
```
Comment (chriddyp) [inline src/components/Interval.react.js]: > Number of times the interval will be fired. If -1, then the interval has no limit (the default) and if 0 then the interval stops running.

File: test/test_integration.py [hunk 2]
```diff
@@ -528,3 +528,22 @@ def update_graph(n_clicks):
...
+        ])
+
+        @app.callback(Output('output', 'children'),
+                    [Input('interval', 'n_intervals')])
+        def update_text(n):
+            return "{}".format(n)
+
+        self.startServer(app=app)
+
+        time.sleep(2) # wait for interval to finish
```
Comment (chriddyp) [inline test/test_integration.py]: I'd make this like `time.sleep(5)` just to make sure that you didn't catch it on `2` "by chance"

**Pull request merged**